tests/components/dlna_dmr/test_config_flow.py

```diff
@@ -515,28 +447,40 @@ async def test_ssdp_flow_success(hass: HomeAssistant) -> None:
 async def test_ssdp_flow_unavailable(
     hass: HomeAssistant, domain_data_mock: Mock
 ) -> None:
-    """Test that SSDP discovery with an unavailable device gives an error message.
+    """Test that SSDP discovery with an unavailable device still succeeds.
 
-    This may occur if the device is turned on, discovered, then turned off
-    before the user attempts to add it.
+    All the required information for configuration is obtained from the SSDP
+    message, there's no need to connect to the device to configure it.
     """
     result = await hass.config_entries.flow.async_init(
         DLNA_DOMAIN,
         context={"source": config_entries.SOURCE_SSDP},
         data=MOCK_DISCOVERY,
     )
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["errors"] == {}
     assert result["step_id"] == "confirm"
 
     domain_data_mock.upnp_factory.async_create_device.side_effect = UpnpError
 
     result = await hass.config_entries.flow.async_configure(
         result["flow_id"], user_input={}
     )
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["errors"] == {"base": "could_not_connect"}
-    assert result["step_id"] == "confirm"
+    await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == MOCK_DEVICE_NAME
+    assert result["data"] == {
+        CONF_URL: MOCK_DEVICE_LOCATION,
+        CONF_DEVICE_ID: MOCK_DEVICE_UDN,
+        CONF_TYPE: MOCK_DEVICE_TYPE,
+    }
+    assert result["options"] == {}
+
+    # Remove the device to clean up all resources, completing its life cycle
+    entry_id = result["result"].entry_id
+    assert await hass.config_entries.async_remove(entry_id) == {
```

Comment:
We don't need to remove the config entry in the config flow tests. If we want to test that it's normally part of the `test_init.p` module tests.

Reply:
In this case it's to satisfy the post-conditions in the `domain_data_mock` and `dmr_device_mock` fixtures. The post-conditions check that all acquired resources have been released.

I'd prefer to keep the post-conditions and the clean up in, as it has already managed to find bugs for me while I was writing the `DlnaDmrEntity` code. I can put the clean up into a helper function to make it more obvious what's going on, though.